src/main/java/bio/terra/service/dataset/DatasetBucketDao.java

```diff
@@ -85,7 +88,7 @@ public void decrementDatasetBucketLink(UUID datasetId, UUID bucketResourceId) {
         datasetBucketLinkUpdate(sqlDecrementCount, datasetId, bucketResourceId);
     }
 
-    private boolean datasetBucketLinkExists(UUID datasetId, UUID bucketResourceId) {
+    boolean datasetBucketLinkExists(UUID datasetId, UUID bucketResourceId) {
```

Comment:
followon note: we may want to pull on the @VisibleForTesting / @RestrictedApiChecker thread for cases like this where we don't really want the method to be non-private. I wouldn't do that as part of this PR though